src/bin/calculate_offsets/ethereum/rfc003/templates/ether/contract.asm

```diff
@@ -45,8 +45,9 @@
     redeem
     jumpi
 
-    // Exit if hashes don't match
-    return(0, 0)
+    // Jump to invalid secret if no match
+    invalid_secret
+    jump
```

Comment:
Maybe not possible but can you replace this with the following:
```
invalid_secret:
    // return "invalidSecret" = 0x696e76616c69645365637265740000000000000000000000000000000000000000
    mstore(0, "invalidSecret")
    revert(0, 32)
```

Il L46 `jumpi` does not jump, wouldn't it just continue and trigger the invalid secret? This would make the contract shorter and save one jump instruction.

Same applies for ERC20 contract.

Reply:
Do you propose to move the instructions after the label `invalid_secret` inline? 
This is of course possible but we would have some duplications as on line 23 we also need to jump to the same result. 

The contract would be longer but we would get rid of one jump instructions.

Reply:
I expect to keep the label and remove the dupe.

Let's have a chat once you are in office?